cli/medperf/commands/benchmark/submit.py

```diff
@@ -45,97 +41,38 @@ def run(cls, benchmark_info: dict, force_test: bool = True):
-            ),
-            (
-                "reference_model_mlcube",
-                model_uid_valid,
-                "Reference Model MLCube is invalid",
-            ),
-            (
-                "data_evaluator_mlcube",
-                eval_uid_valid,
-                "Data Evaluator MLCube is invalid",
-            ),
-        ]
-
-        valid = True
-        for attr, test, error_msg in valid_tests:
-            if not test:
-                valid = False
-                self.ui.print_error(error_msg)
-
-        return valid
-
     def get_extra_information(self):
         """Retrieves information that must be populated automatically,
         like hash, generated uid and test results
         """
-        tmp_uid = self.demo_hash if self.demo_hash else generate_tmp_uid()
-        demo_dset_path = self.comms.get_benchmark_demo_dataset(self.demo_url, tmp_uid)
+        bmk_demo_url = self.bmk.demo_dataset_tarball_url
+        bmk_demo_hash = self.bmk.demo_dataset_tarball_hash
+        tmp_uid = bmk_demo_hash if bmk_demo_hash else generate_tmp_uid()
+        demo_dset_path = self.comms.get_benchmark_demo_dataset(bmk_demo_url, tmp_uid)
         demo_hash = get_file_sha1(demo_dset_path)
-        if self.demo_hash and demo_hash != self.demo_hash:
-            logging.error(
-                f"Demo dataset hash mismatch: {demo_hash} != {self.demo_hash}"
-            )
+        if bmk_demo_hash and demo_hash != bmk_demo_hash:
+            logging.error(f"Demo dataset hash mismatch: {demo_hash} != {bmk_demo_hash}")
             raise InvalidEntityError(
                 "Demo dataset hash does not match the provided hash"
             )
-        self.demo_hash = demo_hash
+        self.bmk.demo_dataset_tarball_hash = demo_hash
         demo_uid, results = self.run_compatibility_test()
-        self.demo_uid = demo_uid
-        self.results = results
+        self.bmk.demo_dataset_generated_uid = demo_uid
+        self.bmk.metadata = {"results": results}
 
     def run_compatibility_test(self):
         """Runs a compatibility test to ensure elements are compatible,
         and to extract additional information required for submission
         """
         self.ui.print("Running compatibility test")
-        data_prep = self.data_preparation_mlcube
-        model = self.reference_model_mlcube
-        evaluator = self.data_evaluator_mlcube
-        demo_url = self.demo_url
-        demo_hash = self.demo_hash
+        data_prep = self.bmk.data_preparation_mlcube
+        model = self.bmk.reference_model_mlcube
+        evaluator = self.bmk.data_evaluator_mlcube
+        demo_url = self.bmk.demo_dataset_tarball_url
+        demo_hash = self.bmk.demo_dataset_tarball_hash
         benchmark = Benchmark.tmp(data_prep, model, evaluator, demo_url, demo_hash)
         _, data_uid, _, results = CompatibilityTestExecution.run(
             benchmark.generated_uid, force_test=self.force_test
```

Comment:
I am thinking about passing the benchmark directly to the compatibility test rather than creating a tmp one. This can be in another PR though